transactions/views.py

```diff
@@ -85,3 +94,86 @@ def post(self, request):
+                "invest_completed_amount": portfolios.filter(
+                    investment_state_id=InvestmentState.State.COMPLETE.value
+                ).aggregate(amounts=Sum("amounts"))["amounts"],
+                "loss_amount": portfolios.filter(
+                    investment_state_id=InvestmentState.State.LOSS.value
+                ).aggregate(amounts=Sum("amounts"))["amounts"],
+                "investing_normal_amount": portfolio_investings.filter(
+                    repayment_state_id=RepaymentState.State.NORMAL.value
+                ).aggregate(amounts=Sum("amounts"))["amounts"],
+                "investing_delay_amount": portfolio_investings.filter(
+                    repayment_state_id=RepaymentState.State.DELAY.value
+                ).aggregate(amounts=Sum("amounts"))["amounts"],
+                "investing_overdue_amount": portfolio_investings.filter(
+                    repayment_state_id=RepaymentState.State.OVERDUE.value
+                ).aggregate(amounts=Sum("amounts"))["amounts"],
+            },
+            "portfolio_current_condition": {
+                "grade": {
+                    "A": portfolios.filter(
+                        Q(investment__grade__name="A+")
+                        | Q(investment__grade__name="A")
+                        | Q(investment__grade__name="A-")
+                    ).aggregate(amounts=Sum("amounts"))["amounts"],
+                    "B": portfolios.filter(
+                        Q(investment__grade__name="B+")
+                        | Q(investment__grade__name="B")
+                        | Q(investment__grade__name="B-")
+                    ).aggregate(amounts=Sum("amounts"))["amounts"],
+                    "C": portfolios.filter(
+                        Q(investment__grade__name="C+")
+                        | Q(investment__grade__name="C")
+                        | Q(investment__grade__name="C-")
+                    ).aggregate(amounts=Sum("amounts"))["amounts"],
+                    "D": portfolios.filter(
+                        Q(investment__grade__name="D+")
+                        | Q(investment__grade__name="D")
+                        | Q(investment__grade__name="D-")
+                    ).aggregate(amounts=Sum("amounts"))["amounts"],
+                },
+                "return_rate": {
+                    "8_under": portfolios.filter(investment__return_rate__lt=8).aggregate(amounts=Sum("amounts"))["amounts"],
+                    "8_over_or_equal": portfolios.filter(investment__return_rate__gte=8).aggregate(amounts=Sum("amounts"))["amounts"],
+                    "10_over_or_equal": portfolios.filter(investment__return_rate__gte=10).aggregate(amounts=Sum("amounts"))["amounts"],
+                    "12_over_or_equal": portfolios.filter(investment__return_rate__gte=12).aggregate(amounts=Sum("amounts"))["amounts"],
+                }
+            }
+        }
+
+        return JsonResponse({"results": results}, status=200)
```

Comment:
예준님 다음에 리펙토링을 위해서 코멘트 남겨 놓습니다! (지금 반영하지 말고 프로젝트 끝나고 반영해보세요!)

지금은 매번 `aggregate()` 하기에 database에 많은 요청을 보내고 있는 것 같아 보입니다. 아래와 같이 annotate + filter 조합을 사용해서 리펙토링 해보세요~

```suggestion
        user = User.objects.annotate(
            total_amounts_of_grade_a = Sum("protfoilos__amounts", filter=Q(investment__grade__name__contains="A")),
            total_amounts_of_grade_b = Sum("protfoilos__amounts", filter=Q(investment__grade__name__contains="B")),
            ..
        ).get(id = request.user.id)
```